pkg/webhook/create_options.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2020 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package webhook
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	scheme "k8s.io/client-go/kubernetes/scheme"
+	rest "k8s.io/client-go/rest"
+)
+
+var newCreateWithOptions func(client rest.Interface, namespace string) podInterface = newPods
+
+type pods struct {
+	client rest.Interface
+	ns     string
+}
+
+type podInterface interface {
+	createWithOptions(ctx context.Context, pod *corev1.Pod, opts metav1.CreateOptions) (*corev1.Pod, error)
+}
+
+// newPods returns a Pods
+func newPods(client rest.Interface, namespace string) podInterface {
+	p := &pods{
+		client: client,
+		ns:     namespace,
+	}
+	return podInterface(p)
+}
+
+// CreateWithOptions takes the representation of a pod and creates it.
+//Returns the server's representation of the pod, and an error, if there is any.
+func (c *pods) createWithOptions(ctx context.Context, pod *corev1.Pod, opts metav1.CreateOptions) (result *corev1.Pod, err error) {
```

Comment:
I'd say it's probably simpler to just create a single function to do the create and drop all the interfaces and package variables.

Can we also add a note to drop the use of this function when we bump to K8s 1.18 client libs

Reply:
Added a TODO with a link to the issue.

I think we want to keep the interface/variable just so we can unit test this. Unfortunately we can't use the nice fake client this way.

Reply:
ah i see now